### PR TITLE
[Keyboard] Add MNT Reform USB Keyboard (Standalone)

### PR DESCRIPTION
**Pull request by cdc-mkb**
Add MNT Reform USB Keyboard (Standalone)

## Description

* Add new  `mntre` sources in the `keyboards` directory

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* none

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mntre/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN,
+};
+
+#define _____ KC_NO
```
Comment (drashna) [inline keyboards/mntre/keymaps/default/keymap.c:12]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/mntre/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    static const char PROGMEM qmk_logo[] = {
+        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
+        0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
+        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0x00
+    };
+
+    oled_write_P(qmk_logo, false);
+    return true;
+}
+#endif
```
Comment (drashna) [inline keyboards/mntre/keymaps/default/keymap.c:46]:
```suggestion
```

File: keyboards/mntre/mntre.c [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2021 Cedric Vincent (@cdc-mkb)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "mntre.h"
```
Comment (drashna) [inline keyboards/mntre/mntre.c:4]:
```suggestion
#include "mntre.h"

#ifdef OLED_ENABLE
bool oled_task_kb(void) {
    if (!oled_task_user()) { return false; }
    static const char PROGMEM qmk_logo[] = {
        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8A, 0x8B, 0x8C, 0x8D, 0x8E, 0x8F, 0x90, 0x91, 0x92, 0x93, 0x94,
        0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xB2, 0xB3, 0xB4,
        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0xD2, 0xD3, 0xD4, 0x00
    };

    oled_write_P(qmk_logo, false);
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mntre/keymaps/default/keymap.c (hunk 1 above)
Comment (cdc-mkb) [reply keyboards/mntre/keymaps/default/keymap.c]: done

File: keyboards/mntre/keymaps/default/keymap.c (hunk 2 above)
Comment (cdc-mkb) [reply keyboards/mntre/keymaps/default/keymap.c]: done

File: keyboards/mntre/mntre.c (hunk 3 above)
Comment (cdc-mkb) [reply keyboards/mntre/mntre.c:4]: done

**Comment by cdc-mkb** [comment]
Thanks for your review @drashna !  I updated this pull-request.

File: keyboards/mntre/config.h [hunk 4]
```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 Cedric Vincent (@cdc-mkb)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x03EB
+#define PRODUCT_ID   0x2042
```
Comment (drashna) [inline keyboards/mntre/config.h:10]: This is already used, and needs to be changed.

https://www.the-sz.com/products/usbid/index.php?v=0x03EB&p=0x2042&n=

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mntre/config.h (hunk 4 above)
Comment (cdc-mkb) [reply keyboards/mntre/config.h:10]: I reused the same vendor/product IDs as in the original firmware, but I agree with you they must be changed to something less prone to collision.  I checked in QMK and in the USB ID database that the following change should be Ok:
```suggestion
#define VENDOR_ID    0xFEED
#define PRODUCT_ID   0x1302
```

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**